Production/plugins/TauTupleProducer.cc

```diff
@@ -483,15 +498,95 @@ class TauTupleProducer : public edm::EDAnalyzer {
+            tauTuple().ele_full5x5_e2x5Max.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_e2x5Max() : default_value);
+            tauTuple().ele_full5x5_e5x5.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_e5x5() : default_value);
+            tauTuple().ele_full5x5_r9.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_r9() : default_value);
+            tauTuple().ele_full5x5_hcalDepth1OverEcal.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_hcalDepth1OverEcal() : default_value);
+            tauTuple().ele_full5x5_hcalDepth2OverEcal.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_hcalDepth2OverEcal() : default_value);
+            tauTuple().ele_full5x5_hcalDepth1OverEcalBc.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_hcalDepth1OverEcalBc() : default_value);
+            tauTuple().ele_full5x5_hcalDepth2OverEcalBc.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_hcalDepth2OverEcalBc() : default_value);
+            tauTuple().ele_full5x5_eLeft.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_eLeft() : default_value);
+            tauTuple().ele_full5x5_eRight.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_eRight() : default_value);
+            tauTuple().ele_full5x5_eTop.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_eTop() : default_value);
+            tauTuple().ele_full5x5_eBottom.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_eBottom() : default_value);
+            tauTuple().ele_full5x5_e2x5Left.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_e2x5Left() : default_value);
+            tauTuple().ele_full5x5_e2x5Right.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_e2x5Right() : default_value);
+            tauTuple().ele_full5x5_e2x5Top.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_e2x5Top() : default_value);
+            tauTuple().ele_full5x5_e2x5Bottom.push_back(!ele->hasUserFloat("hgcElectronID:sigmaUU") && ele->polarP4().pt() > 5 ? ele->full5x5_e2x5Bottom() : default_value);
+
+            // Only phase2 electrons with !ele->isEB() obtain a value != default_{int_,}value via hasUserFloat() decision
+            tauTuple().ele_hgcal_sigmaUU.push_back(ele->hasUserFloat("hgcElectronID:sigmaUU") ? ele->userFloat("hgcElectronID:sigmaUU") : default_value);
```

Comment:
I suggest defining a lambda function to simplify the code below:
```c++
const auto fillUserFloat = [&ele](std::vector<float>& output, const std::string& name) {
    output.push_back(ele->hasUserFloat("hgcElectronID:sigmaVV") ? ele->userFloat("hgcElectronID:sigmaVV") : default_value);
};

fillUserFloat(tauTuple().ele_hgcal_sigmaUU, "hgcElectronID:sigmaUU");
...
```

Reply:
Done